opba-protocols/xs2a-protocol/src/main/java/de/adorsys/opba/protocol/xs2a/service/xs2a/oauth2/Xs2aEmbeddedPreAuthorization.java

```diff
@@ -0,0 +1,78 @@
+package de.adorsys.opba.protocol.xs2a.service.xs2a.oauth2;
+
+import de.adorsys.opba.protocol.bpmnshared.dto.DtoMapper;
+import de.adorsys.opba.protocol.bpmnshared.service.context.ContextUtil;
+import de.adorsys.opba.protocol.bpmnshared.service.exec.ValidatedExecution;
+import de.adorsys.opba.protocol.xs2a.context.Xs2aContext;
+import de.adorsys.opba.protocol.xs2a.service.dto.ValidatedQueryHeaders;
+import de.adorsys.opba.protocol.xs2a.service.mapper.QueryHeadersMapperTemplate;
+import de.adorsys.opba.protocol.xs2a.service.xs2a.dto.oauth2.Xs2aOauth2Headers;
+import de.adorsys.opba.protocol.xs2a.service.xs2a.dto.oauth2.Xs2aOauth2WithCodeParameters;
+import de.adorsys.opba.protocol.xs2a.util.logresolver.Xs2aLogResolver;
+import de.adorsys.xs2a.adapter.api.EmbeddedPreAuthorisationService;
+import de.adorsys.xs2a.adapter.api.model.AspspScaApproach;
+import de.adorsys.xs2a.adapter.api.model.EmbeddedPreAuthorisationRequest;
+import de.adorsys.xs2a.adapter.api.model.TokenResponse;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.flowable.engine.delegate.DelegateExecution;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service("xs2aEmbeddedPreAuthorization")
+@RequiredArgsConstructor
+public class Xs2aEmbeddedPreAuthorization extends ValidatedExecution<Xs2aContext> {
+
+    private final Extractor extractor;
+    private final EmbeddedPreAuthorisationService embeddedPreAuthorisationService;
+    private final Xs2aLogResolver logResolver = new Xs2aLogResolver(getClass());
+
+    @Override
+    @SneakyThrows
+    protected void doRealExecution(DelegateExecution execution, Xs2aContext context) {
+        logResolver.log("doRealExecution: execution ({}) with context ({})", execution, context);
+
+        ValidatedQueryHeaders<Xs2aOauth2WithCodeParameters, Xs2aOauth2Headers> validated = extractor.forExecution(context);
+
+        logResolver.log("getToken with parameters: {}", validated.getQuery(), validated.getHeaders());
+        EmbeddedPreAuthorisationRequest embeddedPreAuthorisationRequest = new EmbeddedPreAuthorisationRequest();
```

Comment:
`EmbeddedPreAuthorisationRequest` should be created and mapped with `Xs2aEmbeddedPreAuthorization.Extractor` and not manually